src/Symfony/Component/DependencyInjection/Attribute/AutowireIterator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\TypedReference;
+use Symfony\Contracts\Service\Attribute\SubscribedService;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+/**
+ * Autowires an iterator of services based on a tag name or an explicit list of key => service-type pairs.
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireIterator extends Autowire
+{
+    /**
+     * @see ServiceSubscriberInterface::getSubscribedServices()
+     *
+     * @param string|array<string|SubscribedService> $services A tag name or an explicit list of services
+     * @param string|string[]                        $exclude  A service or a list of services to exclude
+     */
+    public function __construct(
+        string|array $services,
+        string $indexAttribute = null,
+        string $defaultIndexMethod = null,
+        string $defaultPriorityMethod = null,
+        string|array $exclude = [],
+        bool $excludeSelf = true,
+    ) {
+        if (\is_string($services)) {
+            parent::__construct(new TaggedIteratorArgument($services, $indexAttribute, $defaultIndexMethod, false, $defaultPriorityMethod, (array) $exclude, $excludeSelf));
+
+            return;
+        }
+
+        $references = [];
+
+        foreach ($services as $key => $type) {
+            $attributes = [];
+
+            if ($type instanceof SubscribedService) {
+                $key = $type->key ?? $key;
+                $attributes = $type->attributes;
```

Comment:
`symfony/dependencies-injection` requires `"symfony/deprecation-contracts": "^2.5|^3", but `attributes`, `nullable` and `type` are not defined on `SubscribedService` in 2.5. 
is this a mistake?